spec/lib/salus/scanners/bandit_spec.rb

```diff
@@ -0,0 +1,437 @@
+
+        info = scanner.report.to_h.fetch(:info)
+        logs = scanner.report.to_h.fetch(:logs)
+        expect(info[:stdout]).not_to be_nil
+        expect(info[:stdout]).not_to be_empty
+        msg = 'Consider possible security implications associated with cPickle module.'
+        expect(logs).to include(msg)
+      end
+    end
+
+    context 'python project with no known vulnerabilities' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_no_vulns") }
+
+      it 'should report a passing scan' do
+        expect(scanner.report.passed?).to eq(true)
+      end
+    end
+
+    context 'python project with unknown skip' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_with_bandit_errors") }
+
+      it 'should report a failing scan' do
+        expect(scanner.report.passed?).to eq(false)
+
+        info = scanner.report.to_h.fetch(:info)
+        expect(info[:stderr]).to include("ERROR\tUnknown test found in profile: B9999999")
+      end
+    end
+
+    context 'different config options' do
+      let(:repo) { Salus::Repo.new("#{py_dir}/python_project_vulns") }
+
+      it 'aggregate option should work' do
+        repo = Salus::Repo.new("#{py_dir}/python_project_vulns2")
+
+        # aggregate by file
+        config_file = "#{py_dir}/salus_configs/aggregate_file.yaml"
+        configs = Salus::Config.new([File.read(config_file)]).scanner_configs['Bandit']
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: configs)
+        scanner.run
+        expect(scanner.report.passed?).to eq(false)
+
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        results = logs['results']
+
+        # results ordered by filename
+        expect(results.size).to eq(7)
+        expect(results[0]['filename']).to eq('./main.py')
+        results[1..-1].each { |r| expect(r['filename']).to eq('./main2.py') }
+
+        # aggregate by vuln
+        config_file = "#{py_dir}/salus_configs/aggregate_vuln.yaml"
+        configs = Salus::Config.new([File.read(config_file)]).scanner_configs['Bandit']
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: configs)
+        scanner.run
+        expect(scanner.report.passed?).to eq(false)
+
+        logs = JSON.parse(scanner.report.to_h[:logs])
+        results = logs['results']
+
+        # results ordered by vulns (test ids)
+        expect(results.size).to eq(7)
+        expect(results[0]['test_id']).to eq('B403')
+        expect(results[1]['test_id']).to eq('B403')
+        expect(results[2]['test_id']).to eq('B301')
+        expect(results[3]['test_id']).to eq('B301')
+        expect(results[4]['test_id']).to eq('B301')
+        expect(results[5]['test_id']).to eq('B105')
+        expect(results[6]['test_id']).to eq('B105')
+      end
+
+      it 'config file option should work' do
+        repo = Salus::Repo.new("#{py_dir}/python_project_vulns")
+        scanner = Salus::Scanners::Bandit.new(repository: repo, config: {})
```

Comment:
Do we need to run the scanner before the config here?

Reply:
no